ttnn/cpp/ttnn/operations/data_movement/indexed_fill/indexed_fill_pybind.cpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+
+#include "indexed_fill_pybind.hpp"
+#include "indexed_fill.hpp"
+#include "ttnn/cpp/pybind11/decorators.hpp"
+
+
+namespace ttnn::operations::data_movement {
+namespace detail {
+namespace py = pybind11;
```

Comment:
Please follow the ccl/all_gather example:
* Remove heavy includes, use pybind_fwd instead
* Move implementation to cpp